portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -28,37 +34,50 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
   JSONArray messageList = new JSONArray();
+  DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    // Sets the JSON array to the three hard messages
+    Query query = new Query("Task").addSort("timestamp", SortDirection.DESCENDING);
+    PreparedQuery results = datastore.prepare(query);
+
     messageList.clear();
-    messageList.add("Hello Naa'Oli!"); 
-    messageList.add("How are you doing Naa'Oli?");
-    messageList.add("Good to see you again Naa'Oli.");
 
-    // Sends the JSON array to the /data servlet
-    response.setContentType("text/html;");
+    for (Entity entity : results.asIterable()) {
+      messageList.add((String)entity.getProperty("Comment"));
+    }
+
+    response.setContentType("application/json;");
     response.getWriter().println(messageList.toString());
   }
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     // Get Input from the Form
     String preproccessedText = request.getParameter("user-input");
-    if(preproccessedText == null) {
+
+    if (request == null) {
```

Comment:
this check seems wrong - you still want to check if `preprocessedText` is null. If you want to check whether `request` is wrong, you need to check it before line 57 where you use it.